src/test_l1_dsAudio.c

```diff
@@ -10045,54 +8741,42 @@ void test_l1_dsAudio_negative_dsAudioAtmosCapsChangeRegisterCB(void) {
  *
  */
 void test_l1_dsAudio_positive_dsGetSupportedARCTypes(void) {
-	// Logging at the start
-	gTestID = 131;
-	UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
-
-	dsError_t result;
-	int types;
-        intptr_t* handle = NULL;
-
-        // allocate memory for the handles
-        handle = (intptr_t *) calloc(gDSAudioNumberOfPorts , sizeof(intptr_t));
-        if(handle == NULL)
-        {
-                UT_LOG_ERROR("Failed to allcoate memory for handlers");
-                return;
-        }
+    // Logging at the start
+    gTestID = 131;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
 
+    dsError_t result;
+    int types;
+    intptr_t handle = null_handle;
 
-	// Step 01: Initialize audio ports
-	result = dsAudioPortInit();
-	UT_ASSERT_EQUAL(result, dsERR_NONE);
+    // Step 01: Initialize audio ports
+    result = dsAudioPortInit();
+    UT_ASSERT_EQUAL_FATAL(result, dsERR_NONE);
 
-	// Step 02: Get the port handle for all supported audio ports
-	for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
-                if(gDSAudioPortConfiguration[i].typeid != dsAUDIOPORT_TYPE_HDMI_ARC)
-                continue;
+    // Step 02: Get the port handle for all supported audio ports
+    for (int i = 0; i < gDSAudioNumberOfPorts; i++) {
+            if(gDSAudioPortConfiguration[i].typeid != dsAUDIOPORT_TYPE_HDMI_ARC)
+                 continue;
 
-		result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, (handle+i));
-             if(gSourceType == 0) {
-		UT_ASSERT_EQUAL(result, dsERR_NONE);
-		UT_ASSERT_NOT_EQUAL(*(handle+i), null_handle);
+            result = dsGetAudioPort(gDSAudioPortConfiguration[i].typeid, gDSAudioPortConfiguration[i].index, &handle);
+            if(gSourceType == 0) {
+                UT_ASSERT_EQUAL(result, dsERR_NONE);
+                UT_ASSERT_NOT_EQUAL(handle, null_handle);
 
-	        // Step 03: supported ARC types of the connected ARC/eARC
-                result = dsGetSupportedARCTypes(*(handle+i), &types);
+                // Step 03: supported ARC types of the connected ARC/eARC
+                result = dsGetSupportedARCTypes(handle, &types);
                 UT_ASSERT_EQUAL(result, dsERR_NONE);
-             } else {
+            } else {
```

Comment:
code never reaches else part, as the ports are filtered at 8758. check source type outside the for loop. check for other arc function

Reply:
done